portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ListObjects/InspectObject.tsx

```diff
@@ -0,0 +1,181 @@
+  inspectPath,
+  volumeName,
+  setErrorSnackMessage,
+}: IInspectObjectProps) => {
+  const onClose = () => closeInspectModalAndRefresh(false);
+  const [isEncrypt, setIsEncrypt] = useState<boolean>(true);
+  const [decryptionKey, setDecryptionKey] = useState<string>("");
+  const [insFileName, setInsFileName] = useState<string>("");
+
+  if (!inspectPath) {
+    return null;
+  }
+  const makeRequest = async (url: string) => {
+    return await fetch(url, { method: "GET" });
+  };
+
+  const performInspect = async () => {
+    const file = encodeURIComponent(inspectPath);
+    const volume = encodeURIComponent(volumeName);
+
+    const urlOfInspectApi = `/api/v1/admin/inspect?volume=${volume}&file=${file}&encrypt=${isEncrypt}`;
+
+    makeRequest(urlOfInspectApi)
+        .then(async (res) => {
+          if (!res.ok) {
+            const resErr: any = await res.json();
+
+            setErrorSnackMessage({
+              errorMessage: resErr.message,
+              detailedError: resErr.code,
+            });
+          }
+          const blob: Blob = await res.blob();
+
+          //@ts-ignore
+          const filename = res.headers.get("content-disposition").split('"')[1];
+          const decryptKey = getCookieValue(filename) || "";
+
+          performDownload(blob, filename);
+          setInsFileName(filename);
+          setDecryptionKey(decryptKey);
+        })
+        .catch((err) => {
+          setErrorSnackMessage(err);
+        });
+  };
+
+  const onCloseDecKeyModal = () => {
+    deleteCookie(insFileName);
+    onClose();
+    setDecryptionKey("");
+  };
+
+  const onSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+  };
+
+  return (
+    <React.Fragment>
+        {!decryptionKey && <ModalWrapper
+          modalOpen={inspectOpen}
+          titleIcon={<InspectMenuIcon />}
+          title={`Inspect Object`}
+          onClose={onClose}
+          >
+            <form
+                noValidate
+                autoComplete="off"
+                onSubmit={(e: React.FormEvent<HTMLFormElement>) => {
+                  onSubmit(e);
+                }}
+            >
+                Would you like to encrypt{" "}
+                <b>{decodeFileName(inspectPath)}</b>? <br />
+                  <FormSwitchWrapper
+                    label={"Encrypt"}
+                    indicatorLabels={["Yes", "No"]}
+                    checked={isEncrypt}
+                    value={"encrypt"}
+                    id="encrypt"
+                    name="encrypt"
+                    onChange={(e) => {
+                      setIsEncrypt(!isEncrypt);
+                    }}
+                    description=""
+                  />
+              <Grid item xs={12} className={classes.modalButtonBar}>
+                <Button
+                    type="submit"
+                    variant="contained"
+                    color="primary"
+                    onClick={performInspect}
+                >
+                  Save
```

Comment:
The button text could be changed to `Inspect`
```suggestion
                  Inspect
```